connection.go

```diff
@@ -1580,7 +1545,7 @@ func (conn *Connection) ServerProtocolInfo() ProtocolInfo {
 // Since 1.10.0
 func (conn *Connection) ClientProtocolInfo() ProtocolInfo {
 	info := clientProtocolInfo.Clone()
-	info.Auth = conn.opts.Auth
+	info.Auth = conn.serverProtocolInfo.Auth
```

Comment:
The value is not same at all. Why do we need to export the `Auth` from the client side at all?